src/S3/CMakeLists.txt

```diff
@@ -5,7 +5,7 @@ target_include_directories(s3
         include
 )
 
-target_link_libraries(s3 PRIVATE brender SDL2::SDL2 harness compile_with_werror)
+target_link_libraries(s3 PRIVATE brender SDL3::SDL3 harness compile_with_werror)
```

Comment:
We need to better separate s3 and other helper libraries from SDL2.

We still want to support SDL2.
Not all platforms support SDL3 currently, or will be ported to SDL3.
e.g. UWP support ended in upstream SDL3, which is useful for Xbox Series homebrew. There's also an unofficial Switch port that does not yet have a SDL3 port. (Though, if you have an official Nintendo Switch SDK, and are under an NDA, you can get an "official" SDL3 port)